File: app/src/main/java/com/example/injection/component/ApplicationComponent.java
package com.example.injection.component;

import com.example.injection.module.AndroidModule;
import com.example.injection.module.ServicesModule;
import com.example.main.activity.StackRXActivity;
import com.example.main.fragment.QuestionsFragment;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Dependency injection component for the Singleton / Application scope
 */
@Singleton
@Component(modules = {AndroidModule.class, ServicesModule.class})
public interface ApplicationComponent {
    void inject(StackRXActivity stackRXActivity);

    void inject(QuestionsFragment questionsFragment);
}

File: app/src/main/java/com/example/injection/component/ApplicationComponent.kt
package com.example.injection.component

import com.example.injection.module.AndroidModule
import com.example.injection.module.ServicesModule
import com.example.main.activity.StackRXActivity
import com.example.main.fragment.QuestionsFragment
import dagger.Component
import javax.inject.Singleton

/**
 * Dependency injection component for the Singleton / Application scope
 */
@Singleton
@Component(modules = arrayOf(AndroidModule::class, ServicesModule::class))
interface ApplicationComponent {
    fun inject(stackRXActivity: StackRXActivity)
    fun inject(questionsFragment: QuestionsFragment)
}
